### PR TITLE
bug 1216233 Set the HSTS header

### PR DESCRIPTION
**Pull request by jhford**

### CODE REVIEW

**Comment by jhford** [comment]
so we could similar to the trustProxy setting of sslify and use hsts's setIf parameter to see if the x-forward-proto or whatever it's called is SSL and set HSTS on that value, but really, if we're forcing SSL with sslify we should just set hsts!

File: src/app.js [hunk 1]
```diff
@@ -85,6 +86,17 @@ var app = function(options) {
     app.use(sslify.HTTPS(options.trustProxy));
   }
 
+  // When we force SSL, we also want to set the HSTS header file correctly.  We
+  // also want to allow testing code to check for the HSTS header being
+  // generated correctly without having to generate an SSL cert and key and
+  // have express listen on ssl
+  if (options.forceSSL || options.forceHSTS) {
```
Comment (jonasfj) [inline src/app.js:93]: Can we remove the `options.forceHSTS` and just default to having it on if `forceSSL`...

You can still test it using `forceSSL` and `trustProxy`.

**Comment by jonasfj** [comment]
generally this looks good :)

Thanks,

**Comment by jhford** [comment]
forceHSTS is only for testing, because without it we'd need to set up https for testing, if we wanted hsts to be set properly.

**Comment by jonasfj** [comment]
Ah, okay... You could avoid `forceHSTS` but setting `trustProxy` and sending the header that a reverse proxy would attach... anyways, let's land this.

**Pull request merged**